src/averageAges.js

```diff
@@ -15,11 +15,27 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  let counter = 0;
+
+  const totalAge = people.reduce((sum, person) => {
+    if (person.sex === 'm' && century === undefined) {
+      counter++;
+
+      return sum + (person.died - person.born);
+    }
+
+    if (person.sex === 'm'
+      && century
+      && Math.ceil(person.died / 100) === century) {
+      counter++;
+
+      return sum + (person.died - person.born);
+    }
+
+    return sum;
+  }, 0);
+
+  return totalAge / counter;
```

Comment:
Interesting solution and use of 'reduce'.
But it's very complicated.
Can you do it in a more simple way?
You can just 'filter' the incoming 'people' array getting the required list of men.
After that, you need to 'map' this array to get the array of 'ages'
And last of all - calculate average ages and use the length of the array instead of the 'counter'.